src/routes/api.ts

```diff
@@ -1,97 +1,194 @@
+                res.json({
+                    success: false,
+                    result: "Invalid passphrase"
+                });
+            }
         }
     });
 
     router.post("/keys", async (req, res) => {
         const { passphrase } = req.body;
         const keyType: KeyType = req.body.keyType;
-        const key = await context.cckey[keyType].createKey({
-            passphrase
-        });
-        res.json({
-            success: true,
-            result: key
-        });
+
+        while (true) {
+            const priv = generatePrivateKey();
+            const pub = getPublicFromPrivate(priv);
+            const address = getAccountIdFromPublic(pub);
+
+            // Check duplication
+            if ((await findKey(keyType, address)) != null) {
+                continue;
+            }
+
+            let typeId: KeyTypeId | undefined;
+            switch (keyType) {
+                case "asset": {
+                    typeId = KeyTypeId.Asset;
+                    break;
+                }
+                case "platform": {
+                    typeId = KeyTypeId.Platform;
+                    break;
+                }
+            }
+
+            const secret = await storage.encode(
+                priv,
+                typeId!,
+                passphrase,
+                JSON.stringify({})
+            );
+            await KeyModel.query().insert({
+                type: keyType,
+                address: secret.address,
+                version: secret.version,
+
+                kdf: secret.crypto.kdf,
+                kdfparams: secret.crypto.kdfparams,
+                mac: secret.crypto.mac,
+
+                cipher: secret.crypto.cipher,
+                cipherparams: secret.crypto.cipherparams,
+                ciphertext: secret.crypto.ciphertext,
+
+                meta: secret.meta
+            });
+
+            res.json({
+                success: true,
+                result: secret.address
+            });
+            return;
+        }
     });
 
     router.delete("/keys/:key", async (req, res) => {
         const { key } = req.params;
         const keyType: KeyType = req.body.keyType;
-        const result = await context.cckey[keyType].deleteKey({
-            key
-        });
-        res.json({
-            success: true,
-            result
-        });
-    });
 
-    router.post("/keys/:key/sign", async (req, res) => {
-        try {
-            const { key } = req.params;
-            const { message, passphrase = "" } = req.body;
-            const keyType: KeyType = req.body.keyType;
-            const result = await context.cckey[keyType].sign({
-                key,
-                passphrase,
-                message
-            });
+        const row = await findKey(keyType, key);
+
+        if (row == null) {
             res.json({
-                success: true,
-                result
+                success: false,
+                result: "Not found"
```

Comment:
This would be changing the API.
Did you check the SDK about whether it is safe to change the behaviour?

Reply:
All relevant logic are implemented in [this file](https://github.com/CodeChain-io/codechain-sdk-js/blob/master/src/key/RemoteKeyStore.ts).
It seems that I should change `result` to `error`.